### ISSUE TITLE
[master-preview] Composite lumber Feedback

### ISSUE THREAD

**Issue by SlayksWood** [StrangeLoopGames/EcoIssues#13109]
1. FlatRoof connection between themselves. Sometimes look strangely:
![image](https://user-images.githubusercontent.com/45708377/65015943-a6fe1000-d92b-11e9-871c-34c05549f1ce.png)
![image](https://user-images.githubusercontent.com/45708377/65015961-b4b39580-d92b-11e9-837e-b41854fca435.png)
I don’t like blocks that go beyond. But I can do interesting things with such a block. Therefore, this block can be left as an exception. But for another basic blocks (Floor and FullWall) It seems bad:
![image](https://user-images.githubusercontent.com/45708377/65016789-c39b4780-d92d-11e9-9468-824f783efdde.png)
![image](https://user-images.githubusercontent.com/45708377/65016863-f8a79a00-d92d-11e9-9786-fc7245b2385a.png)
Framing should not protrude beyond the block size (1x1x1) or it should have connect too

2. CladWall shapowplacement:
![image](https://user-images.githubusercontent.com/45708377/65013466-dad53780-d923-11e9-89e1-f1450c8deff1.png)

3. Wall trim = Wall:
![image](https://user-images.githubusercontent.com/45708377/65013622-82eb0080-d924-11e9-9303-c553f9ef308e.png)

4. Window, WindowGrilles, DoubleWindow shadowplacement:
![image](https://user-images.githubusercontent.com/45708377/65014416-571d4a00-d927-11e9-8b98-6edb4c1546ef.png)
![image](https://user-images.githubusercontent.com/45708377/65014456-74521880-d927-11e9-93db-a29892c185ea.png)
![image](https://user-images.githubusercontent.com/45708377/65014476-82a03480-d927-11e9-893f-02826fa8bf88.png)

5. DoubleWindow shines through ground and other blocks:
![image](https://user-images.githubusercontent.com/45708377/65014700-1245e300-d928-11e9-9388-ae03d967474f.png)
![image](https://user-images.githubusercontent.com/45708377/65017198-c8acc680-d92e-11e9-9a1e-6d1db88ed9db.png)

6. All SlopeBlocks (besides Slope Point) go beyond. If you build smart you can build wondeful building. But because of their oversized, a lot of strange things can be done.
![image](https://user-images.githubusercontent.com/45708377/65017516-a9faff80-d92f-11e9-8d6e-4a562aa27976.png)
Sometimes you can make a paradox. Take SlopeFlat. This block almost fulled lies outside. You can take another block and don’t understand why you can’t put it in place of SlopeFlat
![image](https://user-images.githubusercontent.com/45708377/65017848-5c32c700-d930-11e9-98d3-a0c0d8e842dd.png)
![image](https://user-images.githubusercontent.com/45708377/65017970-a451e980-d930-11e9-8979-989948f6c93c.png)
I think it needs to return to 1x1x1 size. As option:
Drop below SlopeFlat. So that he would be at the top of the 1x1x1 block.
For all other SlopeBlock I think we need to cut line and drop below blocks.
![image](https://user-images.githubusercontent.com/45708377/65018682-59d16c80-d932-11e9-8078-e01f0ae2a84d.png)
As a bonus:
![2019 09 17 SlopeFlat](https://user-images.githubusercontent.com/45708377/65019126-48d52b00-d933-11e9-8c62-a93e9c6a2ef8.gif)

7. ColumnBlock as FlatRoof block. This block can be left as an exception.
![image](https://user-images.githubusercontent.com/45708377/65019389-c600a000-d933-11e9-9197-89a8b022102d.png)

8. FloatStairs is not flush.
![image](https://user-images.githubusercontent.com/45708377/65020589-2395ec00-d936-11e9-839f-b2a704542368.png)
As SlopeBlock need to cut and drop below.

9. Ramp should start from the ground:
![image](https://user-images.githubusercontent.com/45708377/65021287-72905100-d937-11e9-9421-28783f706e54.png)
RampB should be RampA, RampC should be RampB, RampD should be RampC. RampA as SlopeBlock need to cut and drop below.

**Comment by keegan-o** [comment]
For any block that goes beyond the space there will always be a block that is the same but does not go beyond, so it's purely an extra option. I think unless we stick to 100% cubes players will always be able to do weird stuff with blocks. Adding options is about putting trust in the player to play with different options, which is more interesting than trying to restrict them to only nice combinations which will be impossible anyways. Players can always make ugly stuff. The full wall block is is the same as the cube block only with trim, so it just expands the options. However, I do consider the clipping with diagonal blocks you show there to be an issue I will try to resolve. 

Most of the rest of this is as intended, the 'ramp' blocks are not really supposed to be as useful for roads as road blocks are. 

With the slopes and slope flat blocks-- same as with full wall block: you don't have to use them! If you double stack the slope blocks you get a nice full block slope which you can attach to any traditional full block flat roof style. It's about expanding options without limiting people.

I agree that many players will simply want slopes that do not extend beyond the space however, so it's a matter of adding those much simpler blocks-- not removing the more complicated forms. I'll look into doing this for 9.0 so that we're not forcing people to use the roof-style slopes. I was planning on adding it-- in fact I already have these types of basic slope meshes for some sets-- but wasn't sure if it would make it into 9.0 or not.

With the wall trim blocks, you managed to place only the straight wall, which is the only block between the two forms which is the same. All the edges and intersections have the trim. Fixing this relates to a more fundamental re-factor we will probably do at some point in Eco to simply have all wall pieces be rotate-able selectable forms rather than the 'smart walls' they are right now-- but this isn't on the docket for 9.0.

The floating stairs aren't flush so that they connect to more floating stairs above them properly-- but those could potentially be improved with better logic, for sure-- it just is a bit of a can of worms which I already tried out for the stone stairs, but I'll be keeping my eye on it.

Thanks for the detailed feedback, it's still early days for this set and for continued improvement of the forms and hammer with all our new features we've been adding in 9.0!

I'll open a new bug for the preview meshes which is that they do not display both materials in the block.

**Comment by keegan-o** [comment]
Ah, and the double window is a huge bug that has been around for years which I am in talks with programmers about. (it affects many blocks in the game)

**Comment by SlayksWood** [comment]
1. You steel can build a ramp with such block. You just need to take RampD as the first block and lower it below: 
![image](https://user-images.githubusercontent.com/45708377/65114735-d0806f80-d9ef-11e9-9829-2e16dd74d237.png)
On the other hand, no one forbids you to put an item on RampD:
![image](https://user-images.githubusercontent.com/45708377/65118750-1dfedb80-d9f4-11e9-9252-82cdabdfd890.png)
For me it’s better to hang in the air like with RampA:
![image](https://user-images.githubusercontent.com/45708377/65118958-628a7700-d9f4-11e9-8833-daaa4ee5ace7.png)
2. I agree with SlopeBlocks (especially if other similar blocks will added) but I really dislike SlopeFlat. The block must not lie 90% outside the block. This creates a hoax. Not all people focus on beauty in the game. It should be borne in mind that these blocks give an illusion - formally the room is bigger than it is. THis room seems like 5x4x2, but real this room 5x4x1:
![image](https://user-images.githubusercontent.com/45708377/65120395-3a9c1300-d9f6-11e9-9889-e1c516c6c28a.png)
3. All composite lumber basic blocks = Cube Block! For example take Lumber to compare with Composite Lumber:
- build same construction with Floorblock:
![image](https://user-images.githubusercontent.com/45708377/65128875-ed726e00-da02-11e9-8584-d4724e3f6e77.png)
- for lumber fill empty spaces with CubeBlock
![image](https://user-images.githubusercontent.com/45708377/65128331-fc0c5580-da01-11e9-86a1-df0e8f4b8d25.png)
You can draw patterns on the floor with two different blocks.
- for compoiste lumber fill empty spaces with Cube, FullWall, FlatRoof blocks
![image](https://user-images.githubusercontent.com/45708377/65128691-92407b80-da02-11e9-9406-9e708bfb60e4.png)
All blocks look like one

**Comment by keegan-o** [comment]
I agree about the floors for composite lumber needing work-- my first priority for the set was to ensure players can build sleek looking modern buildings-- but we'll be adding in more and more ornamentation to it as we go. Composite Lumber will eventually have the most ornamentation options in the game for a wood set, but it'll take a little while to get there. The individual species skins will definitely each get their own bling. Thanks for the reminder though, that is good to see the comparison with regular Lumber. I'll see what I can do for 9.0, but you can rest assured it will also improve in the future!

**Comment by keegan-o** [comment]
The world object thing is definitely a bug too, we need to figure out how to make the builders see the world objects as solid, and then that will resolve itself.

**Comment by keegan-o** [comment]
![image](https://user-images.githubusercontent.com/14983165/65287236-43284280-daf7-11e9-97d0-73f658b76c31.png)
These will be making it into 9.0 :)
